INSTALL.md

```diff
@@ -3,94 +3,101 @@
+
+You must first install development tools and libraries to your host.
 Assuming Ubuntu:
 
 ```
 $ sudo apt-get update && \
-sudo apt-get install -y build-essential \
-libssl-dev uuid-dev libgpgme11-dev squashfs-tools libseccomp-dev pkg-config
+  sudo apt-get install -y build-essential \
+  libssl-dev uuid-dev libgpgme11-dev squashfs-tools libseccomp-dev pkg-config
 ```
 
 On CentOS/RHEL:
 
 ```
-$ sudo yum groupinstall -y 'Development Tools'
-$ sudo yum install -y openssl-devel libuuid-devel libseccomp-devel
+$ sudo yum groupinstall -y 'Development Tools' && \
+  sudo yum install -y openssl-devel libuuid-devel libseccomp-devel
 ```
-Skip libseccomp-devel on CentOS/RHEL 6.
+On CentOS/RHEL 6 or less, you may skip `libseccomp-devel`.
 
-## Install golang
+## Install Golang
 
 This is one of several ways to [install and configure golang](https://golang.org/doc/install).
 
-First, visit the [golang download page](https://golang.org/dl/) and pick a
-package archive to download.  Copy the link address and download with `wget`.
+First, download the Golang archive to `/tmp/`, then extract the archive to `/usr/local`: (or use other instructions on Go
+[installation page](https://golang.org/doc/install)).
 
 ```
-$ export VERSION=1.11 OS=linux ARCH=amd64
-$ cd /tmp
-$ wget https://dl.google.com/go/go$VERSION.$OS-$ARCH.tar.gz
+$ export VERSION=1.11.4 OS=linux ARCH=amd64  # change this as you need
+
+$ wget -O /tmp/go${VERSION}.${OS}-${ARCH}.tar.gz https://dl.google.com/go/go${VERSION}.${OS}-${ARCH}.tar.gz && \
+  sudo tar -C /usr/local -xzf /tmp/go${VERSION}.${OS}-${ARCH}.tar.gz
 ```
 
-Then extract the archive to `/usr/local` (or use other instructions on go
-installation page).
+Finally, set up your environment for Go:
 
 ```
-$ sudo tar -C /usr/local -xzf go$VERSION.$OS-$ARCH.tar.gz
+$ echo 'export GOPATH=${HOME}/go' >> ~/.bashrc && \
+  echo 'export PATH=/usr/local/go/bin:${PATH}:${GOPATH}/bin' >> ~/.bashrc && \
+  source ~/.bashrc
 ```
 
-Finally, set up your environment for go
+## Clone the repo
+
+Golang is a bit finicky about where things are placed. Here is the correct way
+to build Singularity from source:
 
 ```
-$ echo 'export GOPATH=${HOME}/go' >> ~/.bashrc
-$ echo 'export PATH=/usr/local/go/bin:${PATH}:${GOPATH}/bin' >> ~/.bashrc
-$ source ~/.bashrc
+$ mkdir -p ${GOPATH}/src/github.com/sylabs && \
+  cd ${GOPATH}/src/github.com/sylabs && \
+  git clone https://github.com/sylabs/singularity.git && \
+  cd singularity
 ```
 
-## Clone the repo
-golang is a bit finicky about where things are placed. Here is the correct way
-to build Singularity from source.
+To build a stable version of Singularity, check out a [release tag](https://github.com/sylabs/singularity/tags) before compiling:
 
 ```
-$ mkdir -p $GOPATH/src/github.com/sylabs
-$ cd $GOPATH/src/github.com/sylabs
-$ git clone https://github.com/sylabs/singularity.git
-$ cd singularity
+$ git checkout v3.0.2
 ```
 
 ## Compile the Singularity binary
```

Comment:
> Compile the Singularity binary

Do you mean, `Compile the Singularity Source Code`?
Or just: `Compiling Singularity`?